core/foundation/src/TClassEdit.cxx

```diff
@@ -735,14 +747,12 @@ bool TClassEdit::IsDefAlloc(const char *allocname,
       }
       a.remove_prefix(end);
    } else {
-      size_t end = k.length();
-      if ( (a[end-1] == '*') || a[end]==' ' ) {
-         size_t skipSpace = (a[end] == ' ');
-         if (a.compare(end+skipSpace,constlen,"const") == 0) {
-            end += constlen+skipSpace;
-         }
+      // Deal with a trailing const of the allocated type
```

Comment:
If it's a "trailing" const, shouldn't it be remove_suffix rather than remove_prefix? (And then removeTrailingSpaces)

Reply:
No, the parsing logic compares allocator and allocated value by "eating up" common parts starting from the front. If the allocated value is empty up to `const>`, it's a default allocator. That's why we always `remove_prefix`.

Reply:
Thanks. Maybe the comment could be clarified as "Deal with an initially trailing (now leading)..."